src/v/kafka/utils/BUILD

```diff
@@ -0,0 +1,23 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+
+redpanda_cc_library(
+    name = "utils",
+    srcs = [
+        "txn_reader.cc",
+    ],
+    hdrs = [
+        "txn_reader.h",
+    ],
+    include_prefix = "kafka/utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/kafka/server",
```

Comment:
It feels weird to have utils that depend on the thing, but I don't have a better suggestion at the moment.

Reply:
right, server probably needs to be broken down further.

Reply:
Yeah agreed. Maybe something reader_utils so as not to be confused for generic utils used by the Kafka layer? Don't feel strongly about it though

Reply:
the reader maybe doesn't need to have its own `utils` directory--it can be a target within server like `kafka/server:txn_reader`?

Reply:
It could but logically this is more a property of a client than a server right? It's built with internal APIs but in reality we should have an internal data path for the local core that hosts the partition because that is what everything is doing here